apps/rush-lib/src/logic/RepoStateFile.ts

```diff
@@ -87,16 +97,46 @@ export class RepoStateFile {
    * @param variant - The variant currently being used by Rush.
    */
   public static loadFromFile(jsonFilename: string, variant: string | undefined): RepoStateFile {
-    let repoStateJson: IRepoStateJson | undefined = undefined;
+    let fileContents: string | undefined;
     try {
-      repoStateJson = JsonFile.loadAndValidate(jsonFilename, RepoStateFile._jsonSchema);
+      fileContents = FileSystem.readFile(jsonFilename);
     } catch (error) {
       if (!FileSystem.isNotExistError(error)) {
         throw error;
       }
     }
 
-    return new RepoStateFile(repoStateJson, jsonFilename, variant);
+    let foundMergeConflictMarker: boolean = false;
+    let repoStateJson: IRepoStateJson | undefined = undefined;
+    if (fileContents) {
+      try {
+        repoStateJson = JsonFile.parseString(fileContents);
+      } catch (error) {
+        // Look for a Git merge conflict marker. PNPM gracefully handles merge conflicts in pnpm-lock.yaml,
+        // so a user should be able to just run "rush update" if they get conflicts in pnpm-lock.yaml
+        // and repo-state.json and have Rush update both.
+        for (
+          let nextNewlineIndex: number = 0;
+          nextNewlineIndex > -1;
+          nextNewlineIndex = fileContents.indexOf('\n', nextNewlineIndex + 1)
+        ) {
+          if (fileContents.substr(nextNewlineIndex + 1, 7) === '<<<<<<<') {
+            foundMergeConflictMarker = true;
+            repoStateJson = {
+              preferredVersionsHash: 'INVALID',
+              pnpmShrinkwrapHash: 'INVALID'
```

Comment:
Does this mean that if any new properties are added they'll need to be included here? Though it does make it more clear what's going wrong.